File: practice/anagramGrouping.py
def isAnagram(s1, s2):
    charSet = set(s1)
    for i in s2:
        if i not in charSet:
            return False
    return True


if __name__ == '__main__':
    a = ['bat', 'eat', 'ate', 'tea', 'cat', 'act']

    groups = {}
    for i in a:
        flag = False
        for g in groups:
            if isAnagram(i, g):
                groups[g].append(i)
                flag = True
        if flag == False:
            groups[i] = [i]

    print(groups)        

File: practice/anagramGrouping.cpp
#include<bits/stdc++.h>

using namespace std;

bool isAnagram(string s1, string s2){
    unordered_set<char> charSet(s1.begin(), s1.end());
    for(char ch : s2){
        if(charSet.find(ch) == charSet.end()){
            return false;
        }
    }
    return true;
}

int main(){
    vector<string> a = {"bat", "eat", "ate", "tea", "cat", "act"};

    cout << isAnagram("live", "evils") << endl;
}